### PR TITLE
Don't persist Zygote backend cache

### LINKED ISSUE #69: Zygote context cache incorrectly(?) persists between AD calls
You can see below more objects accumulate into it every call. I'm not an expert but @oxinabox on Slack suggested this might be incorrect:

```julia
julia> let
           ad = AD.ZygoteBackend()
           AD.derivative(ad, exp, 1.)
           display(ad.ruleconfig.context.cache)
           AD.derivative(ad, exp, 1.)
           display(ad.ruleconfig.context.cache)
       end
IdDict{Any, Any} with 3 entries:
  Box(true)    => RefValue{Any}((contents = nothing,))
  Box(2.71828) => RefValue{Any}((contents = nothing,))
  Box(2.71828) => RefValue{Any}((contents = nothing,))
IdDict{Any, Any} with 6 entries:
  Box(2.71828) => RefValue{Any}((contents = nothing,))
  Box(2.71828) => RefValue{Any}((contents = nothing,))
  Box(true)    => RefValue{Any}((contents = nothing,))
  Box(2.71828) => RefValue{Any}((contents = nothing,))
  Box(true)    => RefValue{Any}((contents = nothing,))
  Box(2.71828) => RefValue{Any}((contents = nothing,))
```

I have a non-MWE where this seems to cause an error, but the error is gone if I manually wipe it. Not sure if it could also cause subtly wrong results. Perhaps it should be wiped before or after each pullback call?

### PR DESCRIPTION
**Pull request by ToucheSir**
Fixes #69 and closes #70.

### CODE REVIEW

File: ext/AbstractDifferentiationZygoteExt.jl [hunk 1]
```diff
@@ -7,6 +7,10 @@ else
     using ..Zygote: Zygote
 end
 
-AD.ZygoteBackend() = AD.ReverseRuleConfigBackend(Zygote.ZygoteRuleConfig())
+AD.@primitive function pullback_function(ba::AD.ZygoteBackend, f, xs...)
+    @inline # @primitive doesn't support this before function keyword
```
Comment (devmotion) [inline ext/AbstractDifferentiationZygoteExt.jl:11]: This syntax is not supported on Julia 1.6.

Comment (devmotion) [reply ext/AbstractDifferentiationZygoteExt.jl:11]:
```suggestion
```

File: src/backends.jl [hunk 2]
```diff
@@ -71,4 +71,4 @@ It is a special case of [`ReverseRuleConfigBackend`](@ref).
 !!! note
     To be able to use this backend, you have to load Zygote.
 """
-function ZygoteBackend end
+struct ZygoteBackend <: AbstractReverseMode end
```
Comment (devmotion) [inline src/backends.jl:74]: This also requires changing the docstring and the README which both state that it is just a special `ReverseRuleConfigBackend`.

**Review by devmotion — COMMENTED** [review]
Could we overload `AD.pullback_function(::ReverseRuleConfig{<:Zygote.ZygoteRuleConfig}, f, xs...)` instead of making `ZygoteRuleConfig` a separate backend? I think it is very unfortunatem and unintuitive if `ReverseRuleBackend(ZygoteRuleConfig())` is broken and behaves differently from `ZygoteBackend`.

File: ext/AbstractDifferentiationZygoteExt.jl (hunk 1 above)
Comment (ToucheSir) [reply ext/AbstractDifferentiationZygoteExt.jl:11]: That's odd, I thought I force-pushed a new commit which removed it. Either way, #77 is the way to go.

**Comment by ToucheSir** [comment]
I thought about that, but somehow got it in my mind that it would require an ugly `@invoke` to the more general `ReverseRuleConfigBackend` method. The code duplication in #77 is unfortunate but pretty harmless, so closing this in favour of that.

**Pull request closed without merging**